crates/core/src/operations/update.rs

```diff
@@ -307,66 +304,24 @@ async fn execute(
         },
     ));
 
-    // Perform another projection but instead calculate updated values based on
-    // the predicate value.  If the predicate is true then evalute the user
-    // provided expression otherwise return the original column value
-    //
-    // For each update column a new column with a name of __delta_rs_ + `original name` is created
```

Comment:
I would really appreciate if someone could take a look at this diff — I think it should replicate the exact logic except heavily simplified and it passes all tests, but there could be an edge case I'm missing.

Reply:
Looks fine to me the change, before it was creating these __delta_rs columns but seems unnecessary to first add those columns and then select only those. Doing this in one projection seems much better

Reply:
Believe me when I say that I stared at this for a couple hours before figuring out what it was doing 🥳 should probably give a nice speed increase too, I didn't benchmark the old code but I did benchmark the projection operation in general and it's very expensive